src/platform/datapath_raw_xdp.c

```diff
@@ -1212,6 +1212,13 @@ CxPlatXdpRx(
             FrameBuffer,
             (uint16_t)Buffer->length);
 
+        //
+        // The route has been filled in with the packet's src/dst IP and ETH addresses, so
+        // mark it resolved. This allows stateless sends to be issued without performing
+        // a route lookup.
+        //
+        Packet->Route->State = RouteResolved;
```

Comment:
Is this really causing an issue? Stateless sends don't care route state. The stateless send codepath doesn't call into SendQueueFlush where the route resolution is happening.

Reply:
They do care. For stateless sends, the RX packet's Route is used. It's passed to CxPlatSocketSend, which then barfs because not only is the State of the route not Resolved, but also the LocalLinkLayerAddress and NextHopLinkLayerAddress are not set up. Without those, there's no way to build the stateless packet. I've fixed that by filling in the ETH addresses in CxPlatDpRawParseEthernet.

Reply:
Local and next hop MAC addresses missing is the problem. Route state doesn't need to be changed here because it's not referenced in stateless case.

Reply:
Stateless send doesn't check `Route->State`.

Reply:
It does. Look at CxPlatSocketSend.
CXPLAT_DBG_ASSERT(Route->State == RouteResolved);

Reply:
Please remove this assert and the change to set route state to Resolved unless you just want to set the route state for this assert to pass. This is something leftover from the old code where we get route in allocating sends.

Reply:
Nothing else in stateless send code path reference route resolution state except this assert.

Reply:
Resolved offline. This assert is important: we have the invariant that Route->State implies the src/dest eth/ip addresses in the Route are valid, which is required for sending a packet, so we should keep the assert here and hence keep the State update.